libff/algebra/scalar_multiplication/test_multiexp.cpp

```diff
@@ -135,41 +147,243 @@ template<typename GroupT, multi_exp_method Method> void test_multiexp_inner()
+template<typename GroupT> void test_multiexp_accumulate_buckets()
 {
-    test_multiexp_accumulate_buckets<alt_bn128_G1>(4);
-    test_multiexp_accumulate_buckets<alt_bn128_G1>(16);
-    test_multiexp_accumulate_buckets<alt_bn128_G1>(128);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_normal>(
+        4);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_normal>(
+        16);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_normal>(
+        128);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_special>(
+        4);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_special>(
+        16);
+    test_multiexp_accumulate_buckets_group<GroupT, multi_exp_base_form_special>(
+        128);
 }
 
-TEST(MultiExpTest, TestMultiExpSignedDigitsRound)
+template<typename GroupT> void test_multiexp_signed_digits_round()
 {
-    test_multiexp_signed_digits_round<alt_bn128_G1, false>(0);
-    test_multiexp_signed_digits_round<alt_bn128_G1, true>(0);
-    test_multiexp_signed_digits_round<alt_bn128_G1, false>(1);
-    test_multiexp_signed_digits_round<alt_bn128_G1, true>(1);
-    test_multiexp_signed_digits_round<alt_bn128_G1, false>(2);
-    test_multiexp_signed_digits_round<alt_bn128_G1, true>(2);
-    test_multiexp_signed_digits_round<alt_bn128_G1, false>(3);
-    test_multiexp_signed_digits_round<alt_bn128_G1, true>(3);
+    test_multiexp_signed_digits_round_group<GroupT, multi_exp_base_form_normal>(
+        0);
+    test_multiexp_signed_digits_round_group<
+        GroupT,
+        multi_exp_base_form_special>(0);
+    test_multiexp_signed_digits_round_group<GroupT, multi_exp_base_form_normal>(
+        1);
+    test_multiexp_signed_digits_round_group<
+        GroupT,
+        multi_exp_base_form_special>(1);
+    test_multiexp_signed_digits_round_group<GroupT, multi_exp_base_form_normal>(
+        2);
+    test_multiexp_signed_digits_round_group<
+        GroupT,
+        multi_exp_base_form_special>(2);
+    test_multiexp_signed_digits_round_group<GroupT, multi_exp_base_form_normal>(
+        3);
+    test_multiexp_signed_digits_round_group<
+        GroupT,
+        multi_exp_base_form_special>(3);
 }
 
-TEST(MultiExpTest, TestMultiExpInner)
+// template<typename GroupT, multi_exp_method Method, multi_exp_base_form
+// baseForm>
+template<typename GroupT, multi_exp_method Method, multi_exp_base_form BaseForm>
+void test_multi_exp_config(size_t num_elements)
 {
-    test_multiexp_inner<alt_bn128_G1, multi_exp_method_BDLO12_signed_mixed>();
+    using Field = typename GroupT::scalar_field;
+
+    std::vector<GroupT> base_elements;
+    base_elements.reserve(num_elements);
+    std::vector<Field> scalars;
+    scalars.reserve(num_elements);
+
+    // Create values for sum n*[1]_G + (n-1)*[2]_G + ... + 1*[n]_G
+    // where [x]_G is encoding of x in G. The result should then be
+    GroupT g = GroupT::one();
```

Comment:
> The result should then be
 
 Is this an unfinished comment?

Reply:
Updated the comment.